features/opennms-es-rest/main-module/src/main/java/org/opennms/plugins/elasticsearch/rest/RestClientFactory.java

```diff
@@ -122,13 +131,23 @@ public void setServers(Set<String> servers) {
 	 * @param esusername Optional HTTP username
 	 * @param espassword Optional HTTP password
 	 */
-	public RestClientFactory(String elasticSearchURL, String esusername, String espassword ){
+	public RestClientFactory(String elasticSearchURL, String esusername, String espassword ) throws MalformedURLException {
 
 		// If multiple URLs are specified in a comma-separated string, split them up
-		config = new HttpClientConfig.Builder(Arrays.asList(elasticSearchURL.split(",")))
-				.multiThreaded(true)
-				.defaultCredentials(esusername, espassword)
-				.build();
+		final List<String> targetHosts = Arrays.asList(elasticSearchURL.split(","));
+		final HttpClientConfig.Builder configBuilder = new HttpClientConfig.Builder(targetHosts).multiThreaded(true);
+		if (!Strings.isNullOrEmpty(esusername) && !Strings.isNullOrEmpty(espassword)) {
+			final URL targetUrl = new URL(targetHosts.get(0));
```

Comment:
Why not set preemptive auth for all targets, using a for() loop here? Or am I missing something?
I'm thinking about the case where we are happily forwarding to the first target, but that cluster member goes away and we switch to the second target, and now performance might suffer because we're no longer doing preemptive auth now. This would be difficult to troubleshoot at best.

Reply:
The API only supports setting it for one target. I could loop over all targets, but that would enable it only for the last target. That is just a limitation of the Jest client library in version 2.0.3